File: api/src/utils/sample_data.py
from datetime import datetime, timedelta
import json
from src.models.user import User, db
from src.models.alumni import Alumni
from src.models.event import Event, EventRegistration
from src.models.message import Message, ForumPost
from src.models.job import Job, JobApplication
from src.models.donation import Donation, DonationCampaign

def create_sample_data():
    """Create sample data for the alumni management system"""
    
    # Create sample users
    users_data = [
        {'username': 'john_doe', 'email': 'john.doe@email.com'},
        {'username': 'sarah_johnson', 'email': 'sarah.johnson@email.com'},
        {'username': 'michael_chen', 'email': 'michael.chen@email.com'},
        {'username': 'emily_rodriguez', 'email': 'emily.rodriguez@email.com'},
        {'username': 'david_thompson', 'email': 'david.thompson@email.com'},
        {'username': 'lisa_wang', 'email': 'lisa.wang@email.com'},
        {'username': 'james_wilson', 'email': 'james.wilson@email.com'},
        {'username': 'admin', 'email': 'admin@university.edu'}
    ]
    
    users = []
    for user_data in users_data:
        user = User(username=user_data['username'], email=user_data['email'])
        db.session.add(user)
        users.append(user)
    
    db.session.commit()
    
    # Create sample alumni profiles
    alumni_data = [
        {
            'user_id': users[0].id,
            'first_name': 'John',
            'last_name': 'Doe',
            'graduation_year': 2018,
            'department': 'Computer Science',
            'current_position': 'Software Engineer',
            'current_company': 'Tech Corp',
            'location': 'San Francisco, CA',
            'bio': 'Passionate software engineer with 5+ years of experience.',
            'skills': json.dumps(['Python', 'JavaScript', 'React', 'Node.js']),
            'is_mentor': True
        },
        {
            'user_id': users[1].id,
            'first_name': 'Sarah',
            'last_name': 'Johnson',
            'graduation_year': 2018,
            'department': 'Computer Science',
            'current_position': 'Senior Software Engineer',
            'current_company': 'Google',
            'location': 'San Francisco, CA',
            'bio': 'Senior engineer passionate about AI and machine learning.',
            'skills': json.dumps(['Python', 'Machine Learning', 'TensorFlow', 'AI']),
            'is_mentor': True
        },
        {
            'user_id': users[2].id,
            'first_name': 'Michael',
            'last_name': 'Chen',
            'graduation_year': 2015,
            'department': 'Business Administration',
            'current_position': 'Product Manager',
            'current_company': 'Microsoft',
            'location': 'Seattle, WA',
            'bio': 'Product manager with expertise in enterprise software.',
            'skills': json.dumps(['Product Management', 'Strategy', 'Analytics', 'Leadership']),
            'is_mentor': True
        },
        {
            'user_id': users[3].id,
            'first_name': 'Emily',
            'last_name': 'Rodriguez',
            'graduation_year': 2020,
            'department': 'Marketing',
            'current_position': 'Digital Marketing Manager',
            'current_company': 'Adobe',
            'location': 'Austin, TX',
            'bio': 'Creative marketing professional specializing in digital campaigns.',
            'skills': json.dumps(['Digital Marketing', 'Content Strategy', 'SEO', 'Analytics']),
            'is_mentor': False
        },
        {
            'user_id': users[4].id,
            'first_name': 'David',
            'last_name': 'Thompson',
            'graduation_year': 2012,
            'department': 'Mechanical Engineering',
            'current_position': 'Engineering Director',
            'current_company': 'Tesla',
            'location': 'Palo Alto, CA',
            'bio': 'Engineering leader with expertise in automotive technology.',
            'skills': json.dumps(['Mechanical Engineering', 'Leadership', 'Automotive', 'Innovation']),
            'is_mentor': True
        },
        {
            'user_id': users[5].id,
            'first_name': 'Lisa',
            'last_name': 'Wang',
            'graduation_year': 2019,
            'department': 'Data Science',
            'current_position': 'Data Scientist',
            'current_company': 'Netflix',
            'location': 'Los Gatos, CA',
            'bio': 'Data scientist working on recommendation algorithms.',
            'skills': json.dumps(['Python', 'Machine Learning', 'Statistics', 'SQL']),
            'is_mentor': True
        },
        {
            'user_id': users[6].id,
            'first_name': 'James',
            'last_name': 'Wilson',
            'graduation_year': 2016,
            'department': 'Finance',
            'current_position': 'Investment Analyst',
            'current_company': 'Goldman Sachs',
            'location': 'New York, NY',
            'bio': 'Investment analyst specializing in technology sector.',
            'skills': json.dumps(['Financial Analysis', 'Investment Strategy', 'Risk Management', 'Excel']),
            'is_mentor': False
        }
    ]
    
    for alum_data in alumni_data:
        alumni = Alumni(**alum_data)
        db.session.add(alumni)
    
    db.session.commit()
    
    # Create sample events
    events_data = [
        {
            'title': 'Annual Alumni Reunion 2024',
            'description': 'Join us for our annual reunion celebration with networking, dinner, and entertainment.',
            'event_date': datetime.now() + timedelta(days=30),
            'location': 'University Campus - Main Hall',
            'is_virtual': False,
            'max_attendees': 200,
            'registration_deadline': datetime.now() + timedelta(days=25),
            'organizer_id': users[7].id,
            'event_type': 'reunion'
        },
        {
            'title': 'Tech Industry Networking Night',
            'description': 'Connect with fellow alumni working in the technology industry.',
            'event_date': datetime.now() + timedelta(days=15),
            'location': 'San Francisco - Tech Hub',
            'is_virtual': False,
            'max_attendees': 100,
            'registration_deadline': datetime.now() + timedelta(days=10),
            'organizer_id': users[1].id,
            'event_type': 'networking'
        },
        {
            'title': 'Virtual Career Workshop',
            'description': 'Learn about career advancement strategies and networking tips.',
            'event_date': datetime.now() + timedelta(days=7),
            'location': 'Virtual Event',
            'is_virtual': True,
            'virtual_link': 'https://zoom.us/j/123456789',
            'max_attendees': 50,
            'registration_deadline': datetime.now() + timedelta(days=5),
            'organizer_id': users[7].id,
            'event_type': 'workshop'
        }
    ]
    
    for event_data in events_data:
        event = Event(**event_data)
        db.session.add(event)
    
    db.session.commit()
    
    # Create sample jobs
    jobs_data = [
        {
            'title': 'Senior Software Engineer',
            'company': 'Google',
            'location': 'Mountain View, CA',
            'job_type': 'full-time',
            'salary_min': 150000,
            'salary_max': 200000,
            'description': 'Join our team building next-generation cloud infrastructure.',
            'requirements': 'BS in Computer Science, 5+ years experience, Python/Java expertise',
            'posted_by': users[1].id,
            'application_deadline': datetime.now() + timedelta(days=30),
            'is_remote': False
        },
        {
            'title': 'Product Manager',
            'company': 'Microsoft',
            'location': 'Seattle, WA',
            'job_type': 'full-time',
            'salary_min': 130000,
            'salary_max': 180000,
            'description': 'Lead product strategy for enterprise software solutions.',
            'requirements': 'MBA or equivalent, 3+ years PM experience, technical background',
            'posted_by': users[2].id,
            'application_deadline': datetime.now() + timedelta(days=25),
            'is_remote': True
        },
        {
            'title': 'Data Scientist',
            'company': 'Netflix',
            'location': 'Los Gatos, CA',
            'job_type': 'full-time',
            'salary_min': 140000,
            'salary_max': 190000,
            'description': 'Work on recommendation algorithms and user behavior analysis.',
            'requirements': 'PhD or MS in related field, Python/R expertise, ML experience',
            'posted_by': users[5].id,
            'application_deadline': datetime.now() + timedelta(days=20),
            'is_remote': False
        }
    ]
    
    for job_data in jobs_data:
        job = Job(**job_data)
        db.session.add(job)
    
    db.session.commit()
    
    # Create sample donation campaigns
    campaigns_data = [
        {
            'title': 'Student Scholarship Fund',
            'description': 'Support deserving students with financial assistance for their education.',
            'goal_amount': 100000.0,
            'current_amount': 67500.0,
            'category': 'scholarships',
            'priority': 'high',
            'end_date': datetime.now() + timedelta(days=90)
        },
        {
            'title': 'New Engineering Lab Equipment',
            'description': 'Help modernize our engineering facilities with state-of-the-art equipment.',
            'goal_amount': 250000.0,
            'current_amount': 180000.0,
            'category': 'facilities',
            'priority': 'medium',
            'end_date': datetime.now() + timedelta(days=120)
        },
        {
            'title': 'Alumni Mentorship Program',
            'description': 'Fund the expansion of our mentorship program to reach more students.',
            'goal_amount': 50000.0,
            'current_amount': 32000.0,
            'category': 'programs',
            'priority': 'low',
            'end_date': datetime.now() + timedelta(days=60)
        }
    ]
    
    for campaign_data in campaigns_data:
        campaign = DonationCampaign(**campaign_data)
        db.session.add(campaign)
    
    db.session.commit()
    
    # Create sample forum posts
    forum_posts_data = [
        {
            'author_id': users[1].id,
            'title': 'Career Transition Tips',
            'content': 'I recently made a successful transition from startup to big tech. Happy to share insights!',
            'category': 'career',
            'likes_count': 15,
            'replies_count': 8
        },
        {
            'author_id': users[2].id,
            'title': 'Product Management Resources',
            'content': 'Here are some great resources for aspiring product managers...',
            'category': 'career',
            'likes_count': 12,
            'replies_count': 5
        },
        {
            'author_id': users[3].id,
            'title': 'Marketing Industry Trends',
            'content': 'The digital marketing landscape is evolving rapidly. Let\'s discuss the latest trends.',
            'category': 'industry',
            'likes_count': 8,
            'replies_count': 3
        }
    ]
    
    for post_data in forum_posts_data:
        post = ForumPost(**post_data)
        db.session.add(post)
    
    db.session.commit()
    
    print("Sample data created successfully!")

File: frontend/src/data/sampleData.js
// Sample alumni data for the demo
export const sampleAlumni = [
  {
    id: 1,
    name: "Sarah Johnson",
    graduationYear: 2018,
    degree: "Computer Science",
    department: "Engineering",
    currentPosition: "Senior Software Engineer",
    company: "Google",
    location: "San Francisco, CA",
    email: "sarah.johnson@email.com",
    phone: "+1 (555) 123-4567",
    linkedin: "linkedin.com/in/sarahjohnson",
    bio: "Passionate software engineer with 6+ years of experience in full-stack development. Currently working on cloud infrastructure at Google.",
    skills: ["JavaScript", "Python", "React", "Node.js", "AWS"],
    interests: ["Technology", "Mentoring", "Travel"],
    profileImage: "/api/placeholder/150/150",
    isAvailableForMentoring: true,
    lastActive: "2024-01-15"
  },
  {
    id: 2,
    name: "Michael Chen",
    graduationYear: 2015,
    degree: "Business Administration",
    department: "Business",
    currentPosition: "Product Manager",
    company: "Microsoft",
    location: "Seattle, WA",
    email: "michael.chen@email.com",
    phone: "+1 (555) 234-5678",
    linkedin: "linkedin.com/in/michaelchen",
    bio: "Product management professional with expertise in enterprise software and user experience design.",
    skills: ["Product Management", "UX Design", "Data Analysis", "Agile"],
    interests: ["Product Strategy", "Innovation", "Basketball"],
    profileImage: "/api/placeholder/150/150",
    isAvailableForMentoring: true,
    lastActive: "2024-01-14"
  },
  {
    id: 3,
    name: "Emily Rodriguez",
    graduationYear: 2020,
    degree: "Marketing",
    department: "Business",
    currentPosition: "Digital Marketing Manager",
    company: "Adobe",
    location: "Austin, TX",
    email: "emily.rodriguez@email.com",
    phone: "+1 (555) 345-6789",
    linkedin: "linkedin.com/in/emilyrodriguez",
    bio: "Creative marketing professional specializing in digital campaigns and brand strategy.",
    skills: ["Digital Marketing", "Content Strategy", "SEO", "Social Media"],
    interests: ["Creative Writing", "Photography", "Yoga"],
    profileImage: "/api/placeholder/150/150",
    isAvailableForMentoring: false,
    lastActive: "2024-01-13"
  },
  {
    id: 4,
    name: "David Thompson",
    graduationYear: 2012,
    degree: "Mechanical Engineering",
    department: "Engineering",
    currentPosition: "Engineering Director",
    company: "Tesla",
    location: "Palo Alto, CA",
    email: "david.thompson@email.com",
    phone: "+1 (555) 456-7890",
    linkedin: "linkedin.com/in/davidthompson",
    bio: "Experienced engineering leader with a focus on sustainable technology and electric vehicles.",
    skills: ["Mechanical Engineering", "Team Leadership", "Product Development", "CAD"],
    interests: ["Sustainability", "Electric Vehicles", "Hiking"],
    profileImage: "/api/placeholder/150/150",
    isAvailableForMentoring: true,
    lastActive: "2024-01-12"
  },
  {
    id: 5,
    name: "Lisa Wang",
    graduationYear: 2019,
    degree: "Data Science",
    department: "Computer Science",
    currentPosition: "Data Scientist",
    company: "Netflix",
    location: "Los Gatos, CA",
    email: "lisa.wang@email.com",
    phone: "+1 (555) 567-8901",
    linkedin: "linkedin.com/in/lisawang",
    bio: "Data scientist passionate about machine learning and recommendation systems.",
    skills: ["Python", "Machine Learning", "SQL", "TensorFlow", "Statistics"],
    interests: ["AI/ML", "Data Visualization", "Cooking"],
    profileImage: "/api/placeholder/150/150",
    isAvailableForMentoring: true,
    lastActive: "2024-01-11"
  },
  {
    id: 6,
    name: "James Wilson",
    graduationYear: 2016,
    degree: "Finance",
    department: "Business",
    currentPosition: "Investment Analyst",
    company: "Goldman Sachs",
    location: "New York, NY",
    email: "james.wilson@email.com",
    phone: "+1 (555) 678-9012",
    linkedin: "linkedin.com/in/jameswilson",
    bio: "Finance professional with expertise in investment analysis and portfolio management.",
    skills: ["Financial Analysis", "Investment Strategy", "Risk Management", "Excel"],
    interests: ["Finance", "Economics", "Tennis"],
    profileImage: "/api/placeholder/150/150",
    isAvailableForMentoring: false,
    lastActive: "2024-01-10"
  }
]

// Sample events data
export const sampleEvents = [
  {
    id: 1,
    title: "Annual Alumni Reunion 2024",
    date: "2024-06-15",
    time: "18:00",
    location: "University Campus - Main Hall",
    description: "Join us for our annual reunion celebration with dinner, networking, and campus tours.",
    category: "Reunion",
    attendees: 156,
    maxAttendees: 200,
    isRegistered: false,
    image: "/api/placeholder/400/200"
  },
  {
    id: 2,
    title: "Tech Industry Networking Night",
    date: "2024-03-22",
    time: "19:00",
    location: "San Francisco - Tech Hub",
    description: "Connect with fellow alumni working in the tech industry. Featuring guest speakers from major tech companies.",
    category: "Networking",
    attendees: 89,
    maxAttendees: 100,
    isRegistered: true,
    image: "/api/placeholder/400/200"
  },
  {
    id: 3,
    title: "Career Development Workshop",
    date: "2024-04-10",
    time: "14:00",
    location: "Virtual Event",
    description: "Learn about career advancement strategies, resume building, and interview techniques.",
    category: "Workshop",
    attendees: 234,
    maxAttendees: 300,
    isRegistered: false,
    image: "/api/placeholder/400/200"
  },
  {
    id: 4,
    title: "Alumni Mentorship Program Launch",
    date: "2024-05-05",
    time: "16:00",
    location: "University Campus - Student Center",
    description: "Launch event for our new mentorship program connecting alumni with current students.",
    category: "Mentorship",
    attendees: 67,
    maxAttendees: 150,
    isRegistered: true,
    image: "/api/placeholder/400/200"
  }
]

// Sample messages data
export const sampleMessages = [
  {
    id: 1,
    sender: "Sarah Johnson",
    subject: "Mentorship Opportunity",
    preview: "Hi! I saw your profile and would love to discuss potential mentorship opportunities...",
    timestamp: "2024-01-15T10:30:00Z",
    isRead: false,
    avatar: "/api/placeholder/40/40"
  },
  {
    id: 2,
    sender: "Alumni Association",
    subject: "Upcoming Tech Networking Event",
    preview: "Don't miss our upcoming tech networking event on March 22nd. Register now...",
    timestamp: "2024-01-14T15:45:00Z",
    isRead: true,
    avatar: "/api/placeholder/40/40"
  },
  {
    id: 3,
    sender: "Michael Chen",
    subject: "Product Management Discussion",
    preview: "Thanks for connecting! I'd love to chat about product management trends...",
    timestamp: "2024-01-13T09:15:00Z",
    isRead: false,
    avatar: "/api/placeholder/40/40"
  }
]

// Sample analytics data for admin dashboard
export const sampleAnalytics = {
  totalAlumni: 2847,
  activeMembers: 1923,
  upcomingEvents: 8,
  totalDonations: 156780,
  monthlyGrowth: 12.5,
  engagementRate: 68.3,
  
  graduationYearDistribution: [
    { year: '2020-2024', count: 456 },
    { year: '2015-2019', count: 789 },
    { year: '2010-2014', count: 623 },
    { year: '2005-2009', count: 445 },
    { year: '2000-2004', count: 334 },
    { year: 'Before 2000', count: 200 }
  ],
  
  departmentDistribution: [
    { department: 'Engineering', count: 892, percentage: 31.3 },
    { department: 'Business', count: 654, percentage: 23.0 },
    { department: 'Arts & Sciences', count: 543, percentage: 19.1 },
    { department: 'Medicine', count: 398, percentage: 14.0 },
    { department: 'Law', count: 234, percentage: 8.2 },
    { department: 'Other', count: 126, percentage: 4.4 }
  ],
  
  locationDistribution: [
    { location: 'California', count: 567 },
    { location: 'New York', count: 423 },
    { location: 'Texas', count: 389 },
    { location: 'Washington', count: 234 },
    { location: 'Massachusetts', count: 198 },
    { location: 'Other', count: 1036 }
  ],
  
  monthlyEngagement: [
    { month: 'Jan', logins: 1234, events: 45, messages: 567 },
    { month: 'Feb', logins: 1456, events: 52, messages: 634 },
    { month: 'Mar', logins: 1678, events: 48, messages: 723 },
    { month: 'Apr', logins: 1543, events: 61, messages: 689 },
    { month: 'May', logins: 1789, events: 55, messages: 756 },
    { month: 'Jun', logins: 1923, events: 67, messages: 834 }
  ]
}

